### PR TITLE
docs(js): fix function setup, env vars in code examples, code examples to use `aws-amplify`

### PR DESCRIPTION
**Pull request by josefaidt**
_Issue #, if available:_

ref #4202

_Description of changes:_

- moves function setup to the top of document, giving context for environment variables
- corrects `update function` -> `add function`
- adds full view of `add function` flow, with resource permissions
- updates GraphQL endpoint env var `process.env.API_URL` -> `process.env.API_<YOUR_API_NAME>_GRAPHQLAPIENDPOINTOUTPUT`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/fragments/lib/graphqlapi/graphql-from-node.mdx [hunk 1]
```diff
@@ -10,9 +10,38 @@ type Todo @model {
 
 This API will have operations available for `create`, `read`, `update`, and `delete`. Let's take a look at how to perform both a __query__ as well as a __mutation__ from a Lambda function using Node.js.
 
+First, create a Lambda function with `amplify add function` and make sure to give access to your GraphQL API when prompted for whether you want to grant access to other resources in the project.
```
Comment (siegerts) [inline src/fragments/lib/graphqlapi/graphql-from-node.mdx:13]:
```suggestion
First, create a Lambda function with `amplify add function` and make sure to give access to your GraphQL API when prompted by the CLI to grant access to other resources in the project.
```

File: src/fragments/lib/graphqlapi/graphql-from-node.mdx [hunk 2]
```diff
@@ -10,9 +10,38 @@ type Todo @model {
...
+
+```console
+ay add function
```
Comment (siegerts) [inline src/fragments/lib/graphqlapi/graphql-from-node.mdx:16]:
```suggestion
amplify add function
```

Comment (josefaidt) [reply src/fragments/lib/graphqlapi/graphql-from-node.mdx:16]: 😷 good catch

File: src/fragments/lib/graphqlapi/graphql-from-node.mdx [hunk 3]
```diff
@@ -167,67 +170,66 @@ const graphqlQuery = require('./query.js').mutation;
...
-        req.headers["x-api-key"] = apiKey;
-    } else {
-        const signer = new AWS.Signers.V4(req, "appsync", true);
-        signer.addAuthorization(AWS.config.credentials, AWS.util.date.getDate());
+  const req = new AWS.HttpRequest(appsyncUrl, region);
+  
+  const item = {
+    input: {
+      name: "Lambda Item",
+      description: "Item Generated from Lambda"
```
Comment (sachscode) [inline src/fragments/lib/graphqlapi/graphql-from-node.mdx:178]:
```suggestion
      description: "Item generated by Lambda function"
```

File: src/fragments/lib/graphqlapi/graphql-from-node.mdx [hunk 4]
```diff
@@ -167,67 +170,66 @@ const graphqlQuery = require('./query.js').mutation;
...
+  req.body = JSON.stringify({
+    query: graphqlQuery,
+    operationName: "createTodo",
+    variables: item
+  });
+  
+  if (apiKey) {
+    req.headers["x-api-key"] = apiKey;
+  } else {
+    const signer = new AWS.Signers.V4(req, "appsync", true);
```
Comment (sachscode) [inline src/fragments/lib/graphqlapi/graphql-from-node.mdx:195]: Do we use "appsync" or do we have a constant defined anywhere like  AWS_SERVICES.APPSYNC?

**Review by sachscode — COMMENTED** [review]
Mostly LGTM, just a few comments

**Comment by josefaidt** [comment]
thoughts on changing the example to use `aws-amplify`? Are there any implications we should be aware of?

```js
// amplify/backend/function/<my-function-name>/src/index.js
/* Amplify Params - DO NOT EDIT
API_10084_GRAPHQLAPIENDPOINTOUTPUT
API_10084_GRAPHQLAPIIDOUTPUT
API_10084_GRAPHQLAPIKEYOUTPUT
ENV
REGION
Amplify Params - DO NOT EDIT */
const { Amplify, API } = require('aws-amplify')

const amplifyConfig = {
  aws_appsync_graphqlEndpoint: process.env.API_10084_GRAPHQLAPIENDPOINTOUTPUT,
  aws_appsync_authenticationType: 'API_KEY',
  aws_appsync_apiKey: process.env.API_10084_GRAPHQLAPIKEYOUTPUT,
}

Amplify.configure(amplifyConfig)

const query = /* GraphQL */ `
  query {
    listTodos {
      items {
        name
      }
    }
  }
`

/**
 * @type {import('@types/aws-lambda').APIGatewayProxyHandler}
 */
exports.handler = async (event) => {
  console.log(`EVENT: ${JSON.stringify(event)}`)
  try {
    return {
      statusCode: 200,
      body: JSON.stringify(await API.graphql({ query })),
    }
  } catch (error) {
    return {
      statusCode: 500,
      body: JSON.stringify(error),
    }
  }
}
```

File: src/fragments/lib/graphqlapi/graphql-from-node.mdx [hunk 5]
```diff
@@ -8,18 +8,54 @@ type Todo @model {
...
-const { print } = graphql;
+<!-- due to placeholder text -->
+<!-- prettier-ignore -->
+```js
+const { Amplify, API } = require('aws-amplify')
+
+const amplifyConfig = {
+  aws_appsync_graphqlEndpoint: process.env.API_<your-api-name>_GRAPHQLAPIENDPOINTOUTPUT,
+  aws_appsync_authenticationType: 'API_KEY',
+  aws_appsync_apiKey: process.env.API_<your-api-name>_GRAPHQLAPIKEYOUTPUT,
```
Comment (josefaidt) [inline src/fragments/lib/graphqlapi/graphql-from-node.mdx:52]:
```suggestion
  aws_appsync_region: process.env.REGION,
```

this isn't required, but would it be beneficial to note?

**Comment by josefaidt** [comment]
**NOTE**: this adjustment also removes the example with a custom signature in favor of showcasing IAM authorization using `aws-amplify`. Should we aim to include this as an additional example under the new `IAM Authorization` section to showcase how to manually sign the request?

\cc @renebrandel @undefobj

**Comment by josefaidt** [comment]
closing in favor of #4278

**Pull request closed without merging**